Dynamic Withdraw: Clash with Pathfinder

There is very likely a clash with dynamic withdraw and the pathfinder. I suspect it's rare. At first look it seems that if the pathfinder mech hits the withdraw zone that was created, even though it's not active, then it tries to trigger withdraw logic but fails and crashes the load.

Turning off dynamic withdraw would be a workaround.

[cleaned_output_log.txt](https://github.com/CWolfs/MissionControl/files/3992555/cleaned_output_log.txt)

```
2020-01-03T14:46:03 - Error - EncounterLogic [ERROR] SmartTrigger [Escape Zone Entered] caught an exception while trying to evaluate.
2020-01-03T14:46:03 - Error - EncounterLogic [ERROR] System.NullReferenceException: Object reference not set to an instance of an object
  at BattleTech.Framework.RegionConditional.Evaluate (MessageCenterMessage message, System.String responseName) [0x000cb] in <3a3b490f618743a4be67f60469483e7a>:0 
  at BattleTech.Framework.SmartTriggerResponse.Evaluate (MessageCenterMessage message) [0x0001f] in <3a3b490f618743a4be67f60469483e7a>:0 : System.NullReferenceException: Object reference not set to an instance of an object
  at BattleTech.Framework.RegionConditional.Evaluate (MessageCenterMessage message, System.String responseName) [0x000cb] in <3a3b490f618743a4be67f60469483e7a>:0 
  at BattleTech.Framework.SmartTriggerResponse.Evaluate (MessageCenterMessage message) [0x0001f] in <3a3b490f618743a4be67f60469483e7a>:0 
2020-01-03T14:46:03 - Error - MessageCenter [ERROR] CRITICAL ERROR, PLEASE REPORT:
Delegate OnMessage - Standard for message type OnRegionEntered failed with exception 
Object reference not set to an instance of an object
  at BattleTech.Framework.RegionConditional.Evaluate (MessageCenterMessage message, System.String responseName) [0x000cb] in <3a3b490f618743a4be67f60469483e7a>:0 
  at BattleTech.Framework.SmartTriggerResponse.Evaluate (MessageCenterMessage message) [0x00059] in <3a3b490f618743a4be67f60469483e7a>:0 
  at BattleTech.Framework.SmartTriggerResponse.OnMessage (MessageCenterMessage message) [0x00000] in <3a3b490f618743a4be67f60469483e7a>:0 
  at MessageCenter.SendMessagesForType (MessageCenterMessageType messageType, MessageCenterMessage message) [0x00037] in <3a3b490f618743a4be67f60469483e7a>:0 
```
for

```
2020-01-03T14:45:47 - Log - MissionControl [LOG] [MissionControl] Setting contract 'Redirection'
2020-01-03T14:45:47 - Log - MissionControl [DEBUG] [MissionControl] Difficulty '9' (Skull value '4.5')
2020-01-03T14:45:47 - Log - MissionControl [DEBUG] [MissionControl] Visisble Difficulty '10' (Skull value '5')
2020-01-03T14:45:47 - Log - MissionControl [LOG] [MissionControl] Using AdditionalLances for difficulty '10' (Skull value '5')
2020-01-03T14:45:47 - Log - MissionControl [LOG] [MissionControl] Contract map is 'mapGeneral_sandyMesa_aDes'
2020-01-03T14:45:47 - Log - MissionControl [LOG] [MissionControl] Setting contract type to 'CaptureEscort' and using Encounter Rule of 'CaptureEscortAdditionalBlockersEncounterRules'
```

--- a/src/Core/EncounterResults/PositionRegionResult.cs
+++ b/src/Core/EncounterResults/PositionRegionResult.cs
@@ -66,7 +66,7 @@ namespace MissionControl.Result {
         }
       }
 
-      // Create new mesh from points and set to collisder and mesh filter
+      // Create new mesh from points and set to collider and mesh filter
       MeshCollider collider = regionGo.GetComponent<MeshCollider>();
       MeshFilter mf = regionGo.GetComponent<MeshFilter>();
       Mesh mesh = MeshTools.CreateHexigon(REGION_RADIUS, meshPoints);

--- a/src/Patches/AbstractActorCheckEnteredCellsForRegionsPatch.cs
+++ b/src/Patches/AbstractActorCheckEnteredCellsForRegionsPatch.cs
@@ -1,0 +1,19 @@
+using Harmony;
+
+using BattleTech;
+
+/*
+  Prevents pathfinder from triggering regions
+*/
+namespace MissionControl.Patches {
+  [HarmonyPatch(typeof(AbstractActor), "CheckEnteredCellsForRegions")]
+  public class AbstractActorCheckEnteredCellsForRegionsPatch {
+    public static bool Prefix(Contract __instance) {
+      if (!MissionControl.Instance.IsMCLoadingFinished) {
+        Main.LogDebug("[CheckEnteredCellsForRegions] MC is running so this is a pathfinder check. Ignoring so not to trigger regions.");
+        return false;
+      }
+      return true;
+    }
+  }
+}

